pkg/config/config.go

```diff
@@ -142,6 +142,16 @@ type PluginConfig struct {
 	// Log line longer than the limit will be split into multiple lines. Non-positive
 	// value means no limit.
 	MaxContainerLogLineSize int `toml:"max_container_log_line_size" json:"maxContainerLogSize"`
+	// NoCgroup indicates to disable the cgroup support.
+	// This is useful when the daemon does not have permission to access cgroups.
+	NoCgroup bool `toml:"no_cgroup" json:"noCgroup"`
+	// NoApparmor indicates to disable the apparmor support.
+	// This is useful when the daemon does not have permission to access Apparmor.
+	NoApparmor bool `toml:"no_apparmor" json:"noApparmor"`
+	// RestrictOOMScoreAdj indicates to limit the lower bound of OOMScoreAdj to the daemon's
+	// current OOMScoreADj.
+	// This is useful when the daemon does not have permission to decrease OOMScoreAdj.
+	RestrictOOMScoreAdj bool `toml:"restrict_oom_score_adj" json:"restrictOOMScoreAdj"`
```

Comment:
I believe these are runtime options and will have to start out as deprecated since they should be specified only for runc? runtimes?

Reply:
See documentation location for updates to config options: https://github.com/containerd/cri/blob/master/docs/config.md

Reply:
See: https://github.com/containerd/containerd/blob/130d07edd2e003b48709c0f4d8769be2ac164c7c/runtime/v2/runc/options/oci.pb.go#L57

Reply:
>  they should be specified only for runc?

Why? `OOMScoreAdj` is defined in OCI Runtime Spec: https://github.com/opencontainers/runtime-spec/blob/5806c35637336642129d03657419829569abc5aa/config.md

Reply:
by runc I mean runc compatible runtimes.. as specified by the OCI Runtime Spec.

Reply:
there's a new feature in kube now to support specifying which runtime(s) you want.. which may be one runc with different options.. say pod 1 with defaults and pod 2 rootless :)

Reply:
My changes are for how to generate OCI Runtime Spec `config.json`, not for OCI Runtime CLI flags. Is `runtime/v2/runc/options/oci.pb.go` designed to hold options for the OCI Runtime Spec `config.json` generator?

Reply:
Yes.

sorry it took so long to get back to you was busy on another urgent project...

Reply:
https://github.com/containerd/cri/pull/943

Reply:
Thanks for the info, I see the TOML section `[plugins.cri.containerd.default_runtime.options]` is designed to hold OCI Runtime Spec `config.json` generator options, however, I'm still not sure the v2 runtime itself is designed to modify `config.json` with options specified in `runtime/v2/runc/options/oci.pb.go`..

Maybe we should introduce a new field like `[plugins.cri.containerd.default_runtime.spec_options]` for `config.json` options?

Reply:
number of ways to do it I suppose.. heading to china for kubecon shanghai. WDYT @Random-Liu?